Change Lufa-MS size

kb-elmo opened the pull request:
## Description

Increased the size of the Lufa-MS bootloader to 8k

Reason stated here:  
https://github.com/abcminiuser/lufa/commit/5ada54b43299564d5fa52d62abfa9648a4b6c207#diff-15ef1ada160ea767f8b2d21026d94de985fcc1641b8b83d69377053ccd6bfddf

The new version also fixed the issue of oversized firmware files overwriting parts of the bootloader and thus softbricking the MCU. 
Writing oversized files to the flashdrive still works to an extent (it will show an error if the file is vastly oversized) but it just results in the keyboard not starting up. It can still be put back into bootloader mode afterwards without problems.
Throwing random files into the flash drive (like the MacOS typical .DS_Store) also doesn't hurt the bootloader anymore.

I guess we could remove the warning for the Lufa-MS bootloader now since it should be safe to use? 

Relevant commit:  
https://github.com/abcminiuser/lufa/commit/ba6d9c1a971db3c42bf0b054ebb64f72b3e3ddba#diff-15ef1ada160ea767f8b2d21026d94de985fcc1641b8b83d69377053ccd6bfddf

Changes for the bootloader were tested on a dummy Arduino Pro Micro with the latest Lufa-MS bootloader.

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on bootloader.mk:
```diff
@@ -90,7 +90,7 @@ ifeq ($(strip $(BOOTLOADER)), USBasp)
 ifeq ($(strip $(BOOTLOADER)), lufa-ms)
     OPT_DEFS += -DBOOTLOADER_MS
-    BOOTLOADER_SIZE = 6144
+    BOOTLOADER_SIZE = 8192
```
I'm opposed to changing this outright, because this can cause issues for older boards (such as the tada68).

Instead, I'd say that this should be configurable, and set in the dztech boards.
```suggestion
    BOOTLOADER_SIZE ?= 6144
```

drashna replied:
At least until LUFA is also updated.

kb-elmo replied:
aight

fauxpark replied:
IMO it should be the other way around - the existing boards should have it set to 6144.

kb-elmo commented:
Changed the bootloader to 8k again since it is already in QMK as Fauxpark said here: https://github.com/qmk/qmk_firmware/pull/13017#discussion_r650781601

Also added local overrides for the two boards that use the old 6k bootloader (Tada68 and Cod67)
